### ISSUE TITLE
Can't build: unreachable dep: apache thrift unknown revision

### ISSUE THREAD

**Issue by felix-fabrega** [golang-migrate/migrate#182]
**Describe the Bug**
While building migrate, it fail while fetching the dependencies. 

**Steps to Reproduce**
1. git clone --quiet --branch v4.2.5 https://github.com/golang-migrate/migrate ./migrate && cd migrate
2. go mod download
3. See error

**Expected Behavior**
The dependencies download normaly so I can build.

**Migrate Version**
v4.2.5

**Go Version**
1.12

**Stacktrace**
```
go: finding github.com/gocql/gocql v0.0.0-20181124151448-70385f88b28b
go: finding github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/xdg/stringprep v1.0.0
go: finding github.com/go-stack/stack v1.8.0
go: finding github.com/dhui/dktest v0.3.0
go: finding golang.org/x/sync v0.0.0-20181221193216-37e7f081c4d4
go: finding google.golang.org/genproto v0.0.0-20190108161440-ae2f86662275
go: finding golang.org/x/net v0.0.0-20190108225652-1e06a53dbb7e
go: finding golang.org/x/tools v0.0.0-20190108222858-421f03a57a64
go: finding github.com/pkg/errors v0.8.1
go: finding golang.org/x/net v0.0.0-20181106065722-10aee1819953
go: finding github.com/tidwall/pretty v0.0.0-20180105212114-65a9db5fad51
go: finding git.apache.org/thrift.git v0.0.0-20180924222215-a9235805469b
go: finding git.apache.org/thrift.git v0.0.0-20180902110319-2566ecd5d999
go: finding cloud.google.com/go v0.34.0
go: finding github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5
go: finding golang.org/x/oauth2 v0.0.0-20181203162652-d668ce993890
go: finding golang.org/x/time v0.0.0-20181108054448-85acf8d2951c
go: finding github.com/sirupsen/logrus v1.3.0
go: finding google.golang.org/grpc v1.16.0
go: finding github.com/sirupsen/logrus v1.2.0
go: finding github.com/docker/docker v0.7.3-0.20190103212154-2b7e084dc98b
go: finding golang.org/x/net v0.0.0-20181220203305-927f97764cc3
go: finding github.com/stretchr/testify v1.3.0
go: finding github.com/docker/distribution v2.7.0+incompatible
go: finding gotest.tools v2.2.0+incompatible
go: finding github.com/mongodb/mongo-go-driver v0.1.0
go: finding github.com/davecgh/go-spew v1.1.0
go: finding golang.org/x/sys v0.0.0-20190102155601-82a175fd1598
go: finding github.com/stretchr/objx v0.1.0
go: finding github.com/gogo/protobuf v1.2.0
go: finding github.com/docker/docker v0.7.3-0.20190108045446-77df18c24acf
go: finding google.golang.org/grpc v1.17.0
go: finding github.com/stretchr/objx v0.1.1
go: finding golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/konsorten/go-windows-terminal-sequences v1.0.1
go: finding github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
go: finding golang.org/x/lint v0.0.0-20181026193005-c67002cb31c3
go: finding golang.org/x/crypto v0.0.0-20180904163835-0709b304e793
go: finding golang.org/x/crypto v0.0.0-20190103213133-ff983b9c42bc
go: finding google.golang.org/appengine v1.4.0
go: finding golang.org/x/sys v0.0.0-20190108104531-7fbe1cd0fcc2
go: git.apache.org/thrift.git@v0.0.0-20180924222215-a9235805469b: git fetch -f https://git.apache.org/thrift.git refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /home/ffabrega/Go/go/pkg/mod/cache/vcs/83dba939f95a790e497d565fc4418400145a1a514f955fa052f662d56e920c3e: exit status 128:
	fatal: unable to access 'https://git.apache.org/thrift.git/': Operation timed out after 300025 milliseconds with 0 out of 0 bytes received
go: git.apache.org/thrift.git@v0.0.0-20180902110319-2566ecd5d999: unknown revision 2566ecd5d999
go: error loading module requirements
```

It does the same thing in Docker. Probably some changes made in apache.org/thrift which leads to this.

**Comment by dhui** [comment]
I'm unable to reproduce the issue.

The commits in question are valid:
* https://github.com/apache/thrift/commit/a9235805469b
* https://github.com/apache/thrift/commit/2566ecd5d999

The error message you received suggests a networking issue:
> fatal: unable to access 'https://git.apache.org/thrift.git/': Operation timed out after 300025 milliseconds with 0 out of 0 bytes received

**Comment by felix-fabrega** [comment]
After a test made in several different locations, you're right. It is very weird but I can't fetch apache thrift from my work location. Thanks and sorry for the false issue.

**Comment by aaahrens** [comment]
@felix-fabrega did you eveer find out what the issue was?

**Comment by pureugong** [comment]
> @felix-fabrega did you ever find out what the issue was?
for short term workaround solution, 
I think you can add `replace` like below at the end of `go.mod` file

```
# in go.mod file
replace git.apache.org/thrift.git => github.com/apache/thrift v0.0.0-20180902110319-2566ecd5d999
```

**Comment by felix-fabrega** [comment]
Hey, TBH I never knew. At this time we had some network issue with our provider, and some domains were not reachable from our location (DNS issue?). But the provider fixed its issue and we never saw the issue back (locally). But sometine it seems to still happen on our CI and we don't understand why. Sorry for not being helpful

**Comment by sgleizes** [comment]
> But sometine it seems to still happen on our CI and we don't understand why.

What happened on our CI was a different error:
```
go: git.apache.org/thrift.git@v0.0.0-20180924222215-a9235805469b: git fetch -f origin refs/heads/*:refs/heads/* refs/tags/*:refs/tags/* in /go/pkg/mod/cache/vcs/83dba939f95a790e497d565fc4418400145a1a514f955fa052f662d56e920c3e: exit status 128:
	fatal: unable to access 'https://git.apache.org/thrift.git/': Failed to connect to git.apache.org port 443: Connection timed out
```

This is probably because this repo no longer exists and they migrated to github at some point.
As @pureugong pointed out this has been fixed in `v4.3.1`, upgrading fixed the issue for us.
